satellite/audit/verifier.go

```diff
@@ -535,8 +553,8 @@ OUTER:
 	return err
 }
 
-// checkIfSegmentDeleted checks if stripe's pointer has been deleted since stripe was selected.
-func (verifier *Verifier) checkIfSegmentDeleted(ctx context.Context, segmentPath string, oldPointer *pb.Pointer) (newPointer *pb.Pointer, err error) {
+// checkIfSegmentDeleted checks if path's pointer has been deleted since path was selected.
+func (verifier *Verifier) checkIfSegmentDeleted(ctx context.Context, segmentPath string) (newPointer *pb.Pointer, err error) {
```

Comment:
We need to verify whether it's the same pointer we started with or not otherwise there could be a situation:

```
1. audit starts with pointer X
2. repair repairs pointer Y, and deletes some pieces from storage nodes A and B
3. audit fails for A and B
```

To ensure that this doesn't happen, we need to verify in step: 3. whether `pointer X` is the same we started with, otherwise we have requested nodes A and B to delete the piece and will fail the audit for them.